src/main/java/com/google/devtools/build/lib/packages/StructImpl.java

```diff
@@ -333,10 +327,16 @@ private void printJson(Object value, StringBuilder sb, Location loc, String cont
     }
   }
 
-  private String jsonEscapeString(String string) {
-    return escapeDoubleQuotesAndBackslashesAndNewlines(string)
+  private void jsonEscapeString(StringBuilder stringBuilder, String string, boolean appendColon) {
+    stringBuilder.append("\"");
+    stringBuilder.append(escapeDoubleQuotesAndBackslashesAndNewlines(string)
         .replace("\r", "\\r")
-        .replace("\t", "\\t");
+        .replace("\t", "\\t"));
+    if (appendColon) {
```

Comment:
The appendColon logic has no business being in here, as it can be composed separately an no extra cost, resulting in a simpler function.

Reply:
Done.